src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SnakemakeExpressionParsing.kt

```diff
@@ -158,19 +141,7 @@ class SnakemakeExpressionParsing(context: SnakemakeParserContext) : ExpressionPa
                         } else {
                             // skip dedents while matched, we could have several dedent tokens in a raw
                             // skip dedent while inside current block (indents > 1)
-
-                            while (indents > 1 && !myBuilder.eof()) {
-                                if (atToken(PyTokenTypes.INCONSISTENT_DEDENT)) {
-                                    incorrectUnindentMarker = myBuilder.mark()
-                                    nextToken()
-                                }
-                                if (atToken(PyTokenTypes.DEDENT)) {
-                                    indents--
-                                    nextToken()
-                                } else {
-                                    break
-                                }
-                            }
+                            skipDedents(1) { incorrectUnindentMarker = myBuilder.mark() }
```

Comment:
Theoretically ` incorrectUnindentMarker = myBuilder.mark()` could be called several times and lead to unbalanced markers error, better to refactor this code to grant that we don't introduce possible unbalanced markers error in case of any unexpected PSI or changes in python code. Here possible sources of such errors are:
1) `myBuilder.mark()`  called several times but we keep reference only on one result
2) some other marker declared before `incorrectUnindentMarker` could be closed while `incorrectUnindentMarker` is still opened

Reply:
@sharkovadarya still actual